qa/tasks/cephfs/kernel_mount.py

```diff
@@ -217,7 +218,9 @@ def read_debug_file(self, filename):
             return stdout.getvalue()
         except CommandFailedError:
             if 'no such file or directory' in stderr.getvalue().lower():
-                return None
+                return errno.ENOENT
```

Comment:
Oh dear! We should ideally `stat` filename to check its existence. Could you please fix that (as a separate commit).

So, this would look like

stat('metrics/size')
if ENOENT:
    read from debugfs
if any error except ENOENT:
    return 0

Reply:
The failure log shared in tracker fails with `Not a directory` i.e. ENOTDIR, so it also need to be taken care of

Reply:
Right - so check metrics/size (via stat) and fallback to scrapping debugfs.

Reply:
@vshankar `cannot statx '/sys/kernel/debug/ceph/8a0a848f-0cbc-473d-a1b7-e4f48c6156d9.client5077/metrics/size': Permission denied` in http://qa-proxy.ceph.com/teuthology/dparmar-2022-05-24_09:43:57-fs-master-distro-default-smithi/6847320/teuthology.log. What could be the reason?

Reply:
Even though it wasnt able to `stat` the `metrics/size` file, all the jobs (http://pulpito.front.sepia.ceph.com/dparmar-2022-05-24_09:43:57-fs-master-distro-default-smithi/) passed because on encountering error other than ENOENT, `get_op_read_count()` is returning 0 and as line 20 in test_readahead.py does `assert op_read >= initial_op_read`, it passes with ease as both of the reads are `0`

Reply:
@vshankar http://pulpito.front.sepia.ceph.com/dparmar-2022-05-24_12:19:04-fs-master-distro-default-smithi/. Working as intended.

Reply:
>@vshankar cannot statx '/sys/kernel/debug/ceph/8a0a848f-0cbc-473d-a1b7-e4f48c6156d9.client5077/metrics/size': Permission denied in http://qa-proxy.ceph.com/teuthology/dparmar-2022-05-24_09:43:57-fs-master-distro-default-smithi/6847320/teuthology.log. What could be the reason?

probably needs running with sudo? (read: omit_sudo)

Reply:
@vshankar yeah its fixed, my last comment contains the logs that shows its working fine :)

Reply:
http://pulpito.front.sepia.ceph.com/dparmar-2022-05-24_11:19:26-fs-master-distro-default-smithi/